sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/SelectQueryGenerator.kt

```diff
@@ -50,13 +50,26 @@ class SelectQueryGenerator(private val query: NamedQuery) : QueryGenerator(query
   fun defaultResultTypeFunction(): FunSpec {
     val function = defaultResultTypeFunctionInterface()
         .addModifiers(OVERRIDE)
+    val argNames = mutableSetOf<String>()
```

Comment:
kotlinpoet has an api to facilitate this, you can create a `NameAllocator` here

Reply:
I would've liked to use an import alias here, but I didn't want to start passing the FileSpec.Builder down the hierarchy (especially because so many tests use SelectQueryGenerator directly)

Reply:
Done